packages/auth/src/tenancy/FunctionContext.js

```diff
@@ -0,0 +1,73 @@
+const cls = require("cls-hooked")
+const { newid } = require("../hashing")
+
+const REQUEST_ID_KEY = "requestId"
+
+class FunctionContext {
```

Comment:
I like this abstraction for using `cls`. We will likely start to lean on this more heavily in future for any kind of system wide state like correlation IDs of requests and other production information.

Reply:
This was an idea put forward by @Rory-Powell based on his previous experience implementing tenancy systems, I think it works quite well for things that are supposed to be hidden!